### PR TITLE
pkg/otlp: move to comp/otelcol/otlp

### PR DESCRIPTION
**Pull request by gbbr**
This change moves the pkg/otlp package to comp/otelcol/otlp as it represents the core code for this component. Some other package still rely on it (pkg/status, pkg/serverless and pkg/metadata/host) but this dependency should be removable once those pacakges are migrated to components.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

File: .gitignore [hunk 1]
```diff
@@ -132,7 +132,7 @@ omnibus/resources/agent/msi/cal/packages/
 omnibus/files/sources
 omnibus/.tarball-version
 tools/windows/install-help/cal/packages/
-[Dd]ebug/
+./[Dd]ebug/
```
Comment (liustanley) [inline .gitignore:135]: Is this change intentional?

**Review by liustanley — APPROVED** [review]
LGTM, just one minor comment/question

**Review by mx-psi — APPROVED** [review]

**Review by DarcyRaynerDD — APPROVED** [review]
